terraform/aio/vm.tf

```diff
@@ -89,10 +89,12 @@ resource "null_resource" "kayobe-aio" {
       # current images, /tmp is noexec, so the path must be changed
       script_path = "/home/${var.ssh_username}/start.sh"
     }
-
+    # In Caracal, OVN fails on Rocky 9 hosts with hostnames that contain a
+    # subdomain (a dot in the name). Required here for the Caracal upgrade Job
     inline = [
       "#!/bin/sh",
-      "echo 'connected!'"
+      "echo 'connected!'",
+      "sudo hostnamectl set-hostname $(sed s/.novalocal// /etc/hostname)"
```

Comment:
Is there going to be an issue if we don't reboot the host to apply the new hostname?

Reply:
e.g. deploy an aio without rebooting, then later reboot

Reply:
Not sure, but the CI here and on [this PR](https://github.com/stackhpc/stackhpc-kayobe-config/pull/1233/) will tell us if anything breaks

Reply:
It won't test my example though, will it?

Reply:
> deploy an aio without rebooting, then later reboot

This would be fairly common with AIOs spawned for dev work outside of CI.

Reply:
New plan, adapting the fix-hostname playbook to our needs. Testing [here](https://github.com/stackhpc/stackhpc-kayobe-config/pull/1233/)